File: Source/api/Models/LibraryModel.cs
namespace MachineTestAPI.Models
{
    public class LibraryModel
    {
        public List<BookModel> Books { get; set; }
        public List<AuthorModel> Authors { get; set; }
        public List<BookAuthorAssociationModel> Associations { get; set; }
    }
}

File: Source/front-end/src/app/_models/library-model.ts
import { AuthorModel } from "./author-model";
import { BookAuthorAssociationModel } from "./book-author-association-model";
import { BookModel } from "./book-model";

export class LibraryModel {
  books:   BookModel[];
  authors: AuthorModel[];
  associations: BookAuthorAssociationModel[];

  public constructor(init?: Partial<LibraryModel>) {
    Object.assign(this, init);
    this.authors = [];
    this.books =[];
    this.associations =[];
  }
}
